protoc-gen-openapiv2/internal/genopenapi/types.go

```diff
@@ -157,7 +157,10 @@ type schemaCore struct {
 	Ref     string          `json:"$ref,omitempty"`
 	Example json.RawMessage `json:"example,omitempty"`
 
-	Items *openapiItemsObject `json:"items,omitempty"`
+	// Properties can be recursively defined
+	Properties *openapiSchemaObjectProperties `json:"properties,omitempty"`
```

Comment:
Moved this from the `openapiSchemaObject`

Reply:
I've just been familiarising myself with the changes I made before adding tests and I'm wondering whether or not this field should be moved back to `openapiSchemaObject`, or whether the change I've made is OK.

If we keep this change, we can also remove the workarounds on lines 1041 and 1108.

Thoughts?

Reply:
I'm a little confused, could you elaborate on the reason for moving it? Does it make more sense for the openapiv2 output we're trying to get?

Reply:
So it makes the code easier to handle the edge-cases for the `.google.protobuf.Empty` type, at the minute there are if statements dotted around on lines 1041 and 1108 which could be removed if this change is kept.

However, as a more general point, I don't really understand why `schemaCore` and `openapiSchemaObject` are different types. The only usage of `schemaCore` in a type is in `openapiSchemaObject` and `openapiItemsObject`. As far as the JSON Schema spec goes, `"items"` is another schema (I'm not certain if this holds true for OpenAPI - I know there are a few differences).

If we wanted to keep things really simple, I'd do away with `schemaCore` completely, move the properties into `openapiSchemaObject` and refactor usages as necessary.